src/pq.py

```diff
@@ -0,0 +1,68 @@
+"""Implement a priority queue."""
+import sys
+
+if sys.version_info[0] == 3:
```

Comment:
It is good accounting for different versions, but float('inf') works for both python2/3
it is the math.inf that does not work in both